File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let max = 0
    let left = 0
    let right = height.length -1
    while (right > left){
        let area = (Math.min(height[left],height[right])*(right-left))
          max = Math.max(area,max)
        if (height[left] < height[right]){
            left++
        }
        else{right--}
    }
    return max
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_amount = 0
        left = 0
        right = len(height)-1
        while(right > left):
            current_area = (right - left)*min(height[left],height[right])
            max_amount = max(max_amount,current_area)
            if height[right] < height[left]:
                right=right-1
            else:
                left=left+1
        return max_amount